File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/IniFile.java
package org.dolphinemu.dolphinemu.model;

import androidx.annotation.Keep;

public class IniFile
{
  @Keep
  private long mPointer;

  @Keep
  public IniFile()
  {
    mPointer = newIniFile();
  }

  private static native long newIniFile();

  @Override
  public native void finalize();

  public native boolean loadFile(String filename, boolean keep_current_data);
  public native boolean saveFile(String filename);

  public native void setString(String section, String key, String value);
  public native String getString(String section, String key, String default_value);

  public native boolean delete(String section, String key);
}

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/IniFile.kt
package org.dolphinemu.dolphinemu.model

import androidx.annotation.Keep

class IniFile {
    @Keep
    private val pointer: Long = newIniFile()

    external fun finalize()

    external fun loadFile(filename: String, keepCurrentData: Boolean): Boolean
    external fun saveFile(filename: String): Boolean

    external fun setString(section: String, key: String, value: String)
    external fun getString(section: String, key: String, defaultValue: String): String

    external fun delete(section: String, key: String): Boolean

    private external fun newIniFile(): Long
}
